drivers: usb_dc_mcux_ehci does not compile

**Describe the bug**
usb_dc_mcux_ehci.c does not compile

**To Reproduce**
Steps to reproduce the behavior:

0. cd tests/subsys/usb/device
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt1050_evk ../
3. make
4. See error

**Impact**
CI fails on `tests/subsys/usb/device`, blocks #17236

**Screenshots or console output**
```
In file included from /home/john/zephyr-project/zephyr/include/toolchain.h:39,
                 from /home/john/zephyr-project/zephyr/include/kernel_includes.h:19,
                 from /home/john/zephyr-project/zephyr/soc/arm/nxp_imx/rt/soc.h:24,
                 from /home/john/zephyr-project/zephyr/drivers/usb/device/usb_dc_mcux_ehci.c:7:
/home/john/zephyr-project/zephyr/include/toolchain/gcc.h:27:28: error: expected identifier or '(' before '_Static_assert'
 #define BUILD_ASSERT(EXPR) _Static_assert(EXPR, "")
                            ^~~~~~~~~~~~~~
/home/john/zephyr-project/zephyr/include/kernel.h:4160:2: note: in expansion of macro 'BUILD_ASSERT'
  BUILD_ASSERT(WB_UP(maxsz) >= _MPOOL_MINBLK);   \
  ^~~~~~~~~~~~
/home/john/zephyr-project/zephyr/drivers/usb/device/usb_dc_mcux_ehci.c:42:11: note: in expansion of macro 'K_MEM_POOL_DEFINE'
 __nocache K_MEM_POOL_DEFINE(ep_buf_pool, 16, 1024, EP_BUF_NUMOF_BLOCKS, 4);
           ^~~~~~~~~~~~~~~~~

```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.10.1
 - 2f95f4afe7bc437b3038f9bb1336668527a6baf4

--- a/include/kernel.h
+++ b/include/kernel.h
@@ -4157,7 +4157,6 @@ struct k_mem_pool {
  * @req K-MPOOL-001
  */
 #define K_MEM_POOL_DEFINE(name, minsz, maxsz, nmax, align)		\
-	BUILD_ASSERT(WB_UP(maxsz) >= _MPOOL_MINBLK);			\
 	char __aligned(WB_UP(align)) _mpool_buf_##name[WB_UP(maxsz) * nmax \
 				  + _MPOOL_BITS_SIZE(maxsz, minsz, nmax)]; \
 	struct sys_mem_pool_lvl _mpool_lvls_##name[Z_MPOOL_LVLS(maxsz, minsz)]; \
@@ -4170,7 +4169,8 @@ struct k_mem_pool {
 			.levels = _mpool_lvls_##name,			\
 			.flags = SYS_MEM_POOL_KERNEL			\
 		} \
-	}
+	}; \
+	BUILD_ASSERT(WB_UP(maxsz) >= _MPOOL_MINBLK);
 
 /**
  * @brief Allocate memory from a memory pool.

